src/anemoi/training/diagnostics/mlflow/logger.py

```diff
@@ -426,7 +432,71 @@ def _get_mlflow_run_params(
     def experiment(self) -> MLFlowLogger.experiment:
         if rank_zero_only.rank == 0:
             self.auth.authenticate()
-        return super().experiment
+
+        parent_obj = super()
+        logger_obj = self
+        experiment = super().experiment
+
+        if self._failed_to_offline:
+            self.return_to_online()
+
+        class WrappedExperiment:
+            """Wrap the experiment object to handle connection errors.
+
+            Fails over to offline logging if an error occurs.
+            """
+
+            def __getattr__(self, key: str) -> Any:
+                # Only wrap logging calls
+                if not hasattr(experiment, key) or not key.startswith("log"):
+                    return super().__getattr__(key)
+                if not callable(getattr(experiment, key)):
+                    return getattr(experiment, key)
+
+                @functools.wraps(getattr(experiment, key))
+                def wrapped_method(*args, **kwargs) -> Any:
+                    try:
+                        return getattr(experiment, key)(*args, **kwargs)
+                    except mlflow.MlflowException as e:
+                        LOGGER.warning("An error occurred when calling %s: %s", key, e)
+                        logger_obj.failover_to_offline()
+                        global experiment
+                        experiment = parent_obj.experiment
+                    return getattr(experiment, key)(*args, **kwargs)
+
+                return wrapped_method
+
+        return WrappedExperiment
+
+    def failover_to_offline(self) -> None:
+        """Failover to offline mode."""
+        self._initialized = False
+        self._mlflow_client = MlflowClient(tracking_uri=self._save_dir)
+        self._failed_to_offline = True
+
+    def return_to_online(self) -> None:
+        """Return to online mode."""
+        try:
+            health_check(self._tracking_uri)
+        except ConnectionError:
+            return
+
+        self._initialized = False
+        self._mlflow_client = MlflowClient(tracking_uri=self._tracking_uri)
+        self._failed_to_offline = False
+
+        from anemoi.training.utils.mlflow_sync import MlFlowSync
+
+        try:
+            MlFlowSync(
```

Comment:
This would try to sync the offline part of the run before logging again online? We'd probably need to use the `self._run_id` and recall `_get_mlflow_run_params` so it restarts from the synced run it